docs/guides/benchmarks.md

```diff
@@ -0,0 +1,162 @@
+    return X_train, X_test, y_train, y_test
+
+
+def train_rf(X_train: np.ndarray, y_train: np.ndarray, n_estimators: int, max_depth: int, random_state: int = 42) -> RandomForestClassifier:
+    model = RandomForestClassifier(n_estimators=n_estimators, max_depth=max_depth, random_state=random_state)
+    model.fit(X_train, y_train)
+    return model
+
+
+def accuracy(model: base.BaseEstimator, y_test: np.ndarray, y_pred: np.ndarray) -> float:
+    accuracy = metrics.accuracy_score(y_test, y_pred)
+    return accuracy
+```
+
+## `@benchmark` for single benchmarks
+Now, inside a new file called `benchmarks.py` we define our benchmarks.
+We first encapsulate the benchmark logic into a function, `benchmark_accuracy()` which prepares the data, trains a classifier, and lastly, obtains the accuracy.
+To mark such a function as a benchmark, we apply the `@benchmark` decorator.
+
+```python
+# benchmarks.py
+import nnbench
+from training import prepare_data, train_rf, accuracy
+
+@nnbench.benchmark()
+def benchmark_accuracy(n_estimators: int, max_depth: int, random_state: int) -> float:
+    X_train, X_test, y_train, y_test = prepare_data()
+    rf = train_rf(X_train=X_train, y_train=y_train, n_estimators=n_estimators,
+                  max_depth=max_depth, random_state=random_state)
+    acc = accuracy(model=rf, X_test=X_test, y_test=y_test)
+    return acc
+```
+
+Lastly, we set up a benchmark runner in the `run_benchmarks.py`. There, we supply the parameters (`n_estimators`, `max_depth`, `random_state`) necessary in the function definition. 
+
+```python
+# run_benchmarks.py
+from nnbench import runner
+
+r = runner.BenchmarkRunner()
+result = r.run("./benchmarks.py", params={"n_estimators": 100, "max_depth": 5, "random_state": 42})
+r.report(to='console', result=result)
+```
+
+When we execute the `run_benchmarks.py` we get the following output.
+
+
+```bash
+python run_benchmarks.py  
+
+
+name         value
+--------  --------
+accuracy  0.933333
+```
+
+# `@parametrize` for multiple configuration benchmarks
+
+Sometimes, we are not only interested in the performance of a model for given parameters but want to compare the performance for different configurations. 
+To achieve this, let's modify our `benchmarks.py` to run a parametrized benchmark. 
+We need to replace the decorator with our `@parametrize` configuration and supply the wanted parameter combinations within an iterable.
+
+```python
+# benchmarks.py
+import nnbench
+from training import prepare_data, train_rf, accuracy
+
+@nnbench.parametrize((
+    {"n_estimators": 10, "max_depth": 2},
+    {"n_estimators": 50, "max_depth": 5},
+    {"n_estimators": 100, "max_depth": 10}
+))
+def benchmark_accuracy(n_estimators: int, max_depth: int, random_state: int) -> float:
+    ... #   Our benchmarking logic from above
+```
+
+We still need to supply the remaining arguments to the `BenchmarkRunner()`, in our case the `random_state`.
```

Comment:
I would like a small sentence before this a la "Notice that the parametrization is incomplete, since we did not specify a value for the `random_state` argument".

Then, just name the argument, i.e. "We now need to supply a value for the `random_state` argument of our benchmark."